Automattic/wp-calypso
# Render correct warning message when `! domain.canCurrentUserAddEmail`

#### Proposed Changes

Sometimes, users are able to access email subscriptions without being the subscription owner. This can happen under the following circumstances:

- User A has a mapped domain: example.com
- User A purchases Professional Email for that domain
- User A removes the example.com domain from their account
- User B adds example.com as a mapped domain to their account
- The Professional Email subscription will now show up in Calypso for User B, but they will not be able to manage it

In this scenario, Calypso wouldn't show an error status for the subscription in the email management view `/email/:domain/manage/:site_slug`. This PR changes that. Any time `domain.canCurrentUserAddEmail` is `false`, we now render the `domain.canCurrentUserAddEmailReason` together with an error status.

When D87509-code is merged, `domain.canCurrentUserAddEmail` will have the correct value under the circumstances outlined above.

**Please note** that the screenshots below are taken with D87509-code applied.

| Before | After |
| - | - |
| ![before02](https://user-images.githubusercontent.com/1101677/189870538-e726dc54-aef8-413c-8194-38b6602e72f6.png) | ![after02](https://user-images.githubusercontent.com/1101677/189900009-e95bd2eb-b855-4f59-bc51-832ea10c8eaa.png) |

#### Testing Instructions



1. Apply D87509-code
2. Do one of the following:
2.1.  Follow the steps outlined above to get access to an email subscription without being the owner (you do need an external domain that you can add as a mapped domain 😫)
2.2. Have a site with an email subscription owned by User A. Add User B as an administrator to that site (`/people/new/:site_slug`).
3. Navigate to `/email/:domain/manage/:site_slug` in Calypso
4. Ensure that the `EmailPlanHeader` renders with an orange border and an orange warning message that reads `Can’t manage subscription`
5. Ensure that a warning message about a subscription owner mismatch is rendered

#### Pre-merge Checklist






Related to #67717
